src/Compiler/Driver/fsc.fs

```diff
@@ -513,6 +513,7 @@ let main1
             rangeForErrors = range0
         )
 
+    tcConfigB.cmdLineArgs <- argv |> String.concat " "
```

Comment:
So we need to keep track of the cmd line arguments here. What's the best way to propagate this along the code?

The current mechanism (via globals) forces this to be mutable - which it doesn't have to be, hence I don't like it too much. Alternative would be creating yet another bunch of optional parameters to affected functions all over the compiler - don't like that too much either...

Reply:
Only TcConfigBuilder uses mutable data, since it is a builder.
This information should be propagated to TcConfig or TcGlobals (depending on where it is needed), where things are immutable.

Reply:
No we do propagate it to TcConfig now, that's what's ParseAndCheckInputs uses now.